Backported from xkeyboard-config 1.6. This patch covers the following commits
from git://anongit.freedesktop.org/git/xkeyboard-config:

    c9b5bf98adf95149cb5efec36939eaf5475e774f
    86f35e3a1094db3f82f6e198e8f7494506f4085e
    91da938a16b4dc29df62fa3794d2d046daa2edbe

diff -uNr xkeyboard-config-1.4.orig/keycodes/evdev xkeyboard-config-1.4/keycodes/evdev
--- xkeyboard-config-1.4.orig/keycodes/evdev	2010-08-05 17:35:56.303383157 +0100
+++ xkeyboard-config-1.4/keycodes/evdev	2010-08-05 17:36:41.182874772 +0100
@@ -36,6 +36,7 @@
 	<AD11> = 34;
 	<AD12> = 35;
 	<BKSL> = 51;
+	alias <AC12> = <BKSL>;
 	<RTRN> = 36;
 
 	<CAPS> = 66;
@@ -299,19 +301,9 @@
 	indicator 11 = "Charging";
 
 	alias <ALGR> = <RALT>;
-};
 
-// For japanese 106 keyboard.
-xkb_keycodes "jp106" {
-    include "evdev(evdev)"
-    <HZTG> = 49;
-    <AC12> = 51;
-};
-
-// For brazilian ABNT2 keyboard.
-xkb_keycodes "abnt2" {
-    include "evdev(evdev)"
-    <KPPT> = 129;
+	// For Brazilian ABNT2
+	alias <KPPT> = <I129>;
 };
 
 // PC98
diff -uNr xkeyboard-config-1.4.orig/keycodes/xfree86 xkeyboard-config-1.4/keycodes/xfree86
--- xkeyboard-config-1.4.orig/keycodes/xfree86	2008-06-01 00:43:02.000000000 +0100
+++ xkeyboard-config-1.4/keycodes/xfree86	2010-08-05 17:36:41.182874772 +0100
@@ -12,6 +12,7 @@
 default xkb_keycodes "xfree86" {
     include "xfree86(basic)"
     <BKSL> =  51;
+    alias <AC12> = <BKSL>;
     <LSGT> =  94;
 };
 
@@ -367,6 +368,9 @@
     indicator 3 = "Scroll Lock";
 
     alias <ALGR> = <RALT>;
+
+    // For Brazilian ABNT2
+    alias <KPPT> = <I06>;
 };
 
 // What keyboard is this?
@@ -400,22 +404,6 @@
 };
 
 
-// For japanese 106 keyboard. by tsuka(tsuka@kawalab.dnj.ynu.ac.jp)
-//
-// All of the keycodes here are now in the basic "xfree86" set.
-//
-xkb_keycodes "jp106" {
-    include "xfree86(basic)"
-    <AC12> = 51;
-};
-
-
-// For brazilian ABNT2 keyboard. by Ricardo Y. Igarashi(iga@that.com.br)
-xkb_keycodes "abnt2" {
-    include "xfree86(xfree86)"
-    <KPPT> = 134;
-};
-
 // IBM ThinkPad Z60m/Z60t/Z61m/Z61t
 xkb_keycodes "thinkpadz60" {
     include "xfree86(xfree86)"
diff -uNr xkeyboard-config-1.4.orig/keymap/xfree86 xkeyboard-config-1.4/keymap/xfree86
--- xkeyboard-config-1.4.orig/keymap/xfree86	2008-05-28 23:51:57.000000000 +0100
+++ xkeyboard-config-1.4/keymap/xfree86	2010-08-05 17:36:41.186839837 +0100
@@ -33,7 +33,7 @@
 };
 // us_intl and br by Ricardo Y. Igarashi (iga@that.com.br)
 xkb_keymap "br"	{
-    xkb_keycodes	{ include "xfree86(abnt2)"	};
+    xkb_keycodes	{ include "xfree86"	};
     xkb_types		{ include "default"		};
     xkb_compatibility	{ include "default"		};
     xkb_symbols		{ include "us(pc101)+br"	};
@@ -141,7 +141,7 @@
     xkb_geometry 	{ include "pc(pc102)"		};
 };
 xkb_keymap "jp106" {
-    xkb_keycodes	{ include "xfree86(jp106)"	};
+    xkb_keycodes	{ include "xfree86"	};
     xkb_types		{ include "default"		};
     xkb_compatibility	{ include "default"		};
     xkb_symbols		{ include "jp(jp106)"		};
diff -uNr xkeyboard-config-1.4.orig/rules/base.lists.part xkeyboard-config-1.4/rules/base.lists.part
--- xkeyboard-config-1.4.orig/rules/base.lists.part	2008-09-10 22:53:02.000000000 +0100
+++ xkeyboard-config-1.4/rules/base.lists.part	2010-08-05 17:36:41.186839837 +0100
@@ -12,7 +12,7 @@
 ! $msmodels = microsoft microsoft7000 microsoftpro microsoftprousb microsoftprose
 
 // PC geometries - they have special geometry but symbols are mostly pc105
-! $pcgeometries = abnt2 latitude jp106 kr106
+! $pcgeometries = latitude
 
 // Layouts that provide further specializations for the OLPC
 ! $olpclayouts = af am ara br ca es et it kh kz in mn np ru th tr us 
diff -uNr xkeyboard-config-1.4.orig/rules/base.m_k.part xkeyboard-config-1.4/rules/base.m_k.part
--- xkeyboard-config-1.4.orig/rules/base.m_k.part	2008-06-01 00:43:02.000000000 +0100
+++ xkeyboard-config-1.4/rules/base.m_k.part	2010-08-05 17:36:41.186839837 +0100
@@ -4,8 +4,6 @@
   sun5		=	sun(type5_euro)
   sun6		=	sun(type6_usb)
   pc98		=	xfree98(pc98)
-  abnt2		=	xfree86(abnt2)
-  jp106		=	xfree86(jp106)
   evdev		=	evdev
   olpc		=	evdev
   macintosh_hhk	=	macintosh(hhk)
diff -uNr xkeyboard-config-1.4.orig/rules/base.xml.in xkeyboard-config-1.4/rules/base.xml.in
--- xkeyboard-config-1.4.orig/rules/base.xml.in	2008-09-24 01:02:42.000000000 +0100
+++ xkeyboard-config-1.4/rules/base.xml.in	2010-08-05 17:36:41.186839837 +0100
@@ -88,13 +88,6 @@
     </model>
     <model>
       <configItem>
-        <name>jp106</name>
-        <_description>Japanese 106-key</_description>
-        <vendor>Generic</vendor>
-      </configItem>
-    </model>
-    <model>
-      <configItem>
         <name>pc98</name>
         <_description>PC-98xx Series</_description>
         <vendor>Generic</vendor>
@@ -123,13 +116,6 @@
     </model>
     <model>
       <configItem>
-        <name>abnt2</name>
-        <_description>Brazilian ABNT2</_description>
-        <vendor>Generic</vendor>
-      </configItem>
-    </model>
-    <model>
-      <configItem>
         <name>airkey</name>
         <_description>Acer AirKey V</_description>
         <vendor>Acer</vendor>
@@ -1066,13 +1052,6 @@
     </model>
     <model>
       <configItem>
-	<name>kr106</name>
-	<_description>Korean 106-key</_description>
-	<vendor>Generic</vendor>
-      </configItem>
-    </model>
-    <model>
-      <configItem>
 	<name>evdev</name>
 	<_description>Evdev-managed keyboard</_description>
 	<vendor>Generic</vendor>
diff -uNr xkeyboard-config-1.4.orig/rules/evdev.m_k.part xkeyboard-config-1.4/rules/evdev.m_k.part
--- xkeyboard-config-1.4.orig/rules/evdev.m_k.part	2008-08-16 13:21:07.000000000 +0100
+++ xkeyboard-config-1.4/rules/evdev.m_k.part	2010-08-05 17:36:41.186839837 +0100
@@ -1,4 +1,2 @@
   pc98		=	evdev(pc98)
-  abnt2		=	evdev(abnt2)
-  jp106		=	evdev(jp106)
   *		=	evdev
